server/src/main/java/org/opentosca/toscana/plugins/lifecycle/TransformationLifecycle.java

```diff
@@ -0,0 +1,36 @@
+package org.opentosca.toscana.plugins.lifecycle;
+
+/**
+ The basic interface to implement the TransformationLifecycle
+ */
+public interface TransformationLifecycle {
+    /**
+     This method is intended to check if the  model given in a transformation context
+     (has to be given via a constructor)
+
+     @return True if the vailidation succeeded, false otherwise.
+     Returning false will cause the transformation to fail
+     */
+    boolean checkModel();
+
+    /**
+     Has to implement the prepare lifecycle operations
```

Comment:
What is the plugin supposed to do here? Improve javadoc.

Reply:
Added some notes below this line